pkg/core/mempool/mempool.go

```diff
@@ -408,6 +417,60 @@ func (m Mempool) onGetMempoolTxs(r rpcbus.Request) (bytes.Buffer, error) {
 	return *w, nil
 }
 
+func (m Mempool) onGetMempoolView(r rpcbus.Request) (bytes.Buffer, error) {
+	// If we want a tx with a certain ID, we can simply look it up
+	// directly
+	txs := make([]transactions.Transaction, 0)
+	if len(r.Params.Bytes()) == 32 {
+		hash, err := hex.DecodeString(string(r.Params.Bytes()))
+		if err != nil {
+			return bytes.Buffer{}, err
+		}
+
+		tx := m.verified.Get(hash)
+		if tx == nil {
+			return bytes.Buffer{}, errors.New("tx not found")
+		}
+
+		txs = append(txs, tx)
+	} else {
+		// In other cases, we will range through the hash map and pick out
+		// what we want depending on the filter.
+		txs = m.verified.Clone()
+
+		if len(r.Params.Bytes()) == 1 {
+			txs = filterTxsByType(txs, transactions.TxType(r.Params.Bytes()[0]))
```

Comment:
nitpick: `Clone()`  iterates through all mempool elements. We can have another API `CloneWithFilter(..)` to avoid `filterTxsByType` array shifting:

Just a suggestion:
```go


func (m HashMap) CloneWithFilter(filterType transactions.TxType) []transactions.Transaction {

	r := make([]transactions.Transaction, 0)
	for _, t := range m.data {
		if t.tx.Type() == filterType {
			r = append(r, t.tx)
		}
	}

	return r
}
```

Reply:
Added it